src/core/read/py_fread.cc

```diff
@@ -90,11 +92,85 @@ static py::oobj fread(const py::PKArgs& args) {
   rdr.init_skipblanks( arg_skipblanks);
   rdr.init_columns(    arg_columns);
   rdr.init_tempdir(    arg_tempdir);
+  rdr.init_multisource(arg_multisrc);
   (void) arg_saveto;
   (void) arg_encoding;
 
-  auto multisource = MultiSource::from_args(args, rdr);
-  return multisource.read_all_fread_style(rdr);
+  MultiSource multisource(args, rdr);
+  return multisource.read_single(rdr);
+}
+
+
+
+//------------------------------------------------------------------------------
+// iread() python function
+//------------------------------------------------------------------------------
+
+static const char* doc_iread =
+R"(iread(anysource=None, *, file=None, text=None, cmd=None, url=None,
```

Comment:
btw, what `i` in `iread` means? and what is the main reason to separate `iread` functionality from `fread`?

Reply:
"i" means iterator: the return value is an iterator producing Frames.
The main reason to have this as a separate function is to make a clear distinction of what is returned by each function: `fread` returns a single Frame, `iread` returns an iterator of Frames.

Reply:
I see. Then it could be somehow inconsistent with the existing `fread` function, where `f` stands for file (?) — that is an input parameter, while for `iread` the first letter `i` stands for the output type.

Reply:
"fread" came from R data.table, where it meant "fast & friendly file finagler". Personally, I don't like the word "finagler", so I'd prefer "f" to mean something else. It could be "file" or "frame", or remain unspecified.

For `iread()` the letter `i` does mean "iterator", but it applies both to the input and the output: after all, in order to produce multiple frames in the output, you need to have multiple sources in the input as well.

That being said, I'm open to suggestions how the `iread` function can be called instead (renaming `fread` is impossible at this point though).

Reply:
I see. To me, if a function is called `something_read()`, I would assume this `something` is the input type. Because then we say: read me a file, read me an archive, etc. Here, if `iread` is going to read various types, it could be called `xread` meaning that it could read from extra sources comparing to `fread`, that only reads files.

Reply:
It could also prevent confusion when typed in uppercase, like now the title of this PR is `Iread ...` that first made me think that the function name starts with `L` :)

Reply:
well, `xread` is already reserved for "extract-read", see #1950

Reply:
Also, `iread` reads from same sources as `fread`, except that it allows a multiple of them.

Reply:
I see, then may be `mread`?

Reply:
I considered `mread` as well, but ultimately thought that `iread` would be more familiar to python users: after all, there are many functions like `izip()`, `islice()`, `ifilter()`, `imap()`, etc (some of them from py2.7) which are iterator versions of functions `zip()`, `map()`, etc.  In this sense, `iread` is like an iterator version of `fread`.